### PR TITLE
CP-34942: update mirage stack

### PR DESCRIPTION
**Pull request by psafont**
This change is tested in https://github.com/xapi-project/xs-opam/pull/559. The PR can be merged once the ci in the xs-opam repo passes.

### CODE REVIEW

File: .github/workflows/ocaml-ci.yml [hunk 1]
```diff
@@ -40,5 +40,9 @@ jobs:
           opam exec -- make build
 
       - name: Run tests
-        run: opam exec -- make test
+        run: opam exec -- dune runtest --instrument-with bisect_ppx --force
 
+      - run: bisect-ppx-report send-to Coveralls
+        env:
+          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PULL_REQUEST_NUMBER: ${{ github.event.number }}
```
Comment (lippirk) [inline .github/workflows/ocaml-ci.yml:48]: How do we see the coverage? Can we add a link to the coveralls page to the readme?

File: lib/mux.ml [hunk 2]
```diff
@@ -71,6 +70,7 @@ end = struct
     transport : R.transport;
     outgoing_mutex: Lwt_mutex.t;
     id_to_wakeup : (R.id, R.request_hdr * ((unit, Protocol.Error.t) result Lwt.u) * R.response_body) Hashtbl.t;
+          * (unit, Protocol.Error.t) result Lwt.u
```
Comment (lippirk) [inline lib/mux.ml:88]: Does this even compile..?

File: .github/workflows/ocaml-ci.yml (hunk 1 above)
Comment (psafont) [reply .github/workflows/ocaml-ci.yml:48]: I'll add a link, coverage can be seen at https://coveralls.io/github/xapi-project/nbd

Comment (lippirk) [reply .github/workflows/ocaml-ci.yml:48]: we can also remove the travis build status

**Review by lippirk — APPROVED** [review]

File: lib/mux.ml (hunk 2 above)
Comment (psafont) [reply lib/mux.ml:88]: The mux tests are compiled but don't seem to run. I've changed the setup so they get run... and they compile just fine, strangely.

**Comment by lindig** [comment]
Are these all administrative changes about the CI, formatting, and so on? Or are there any functional changes in here, too?

**Comment by psafont** [comment]
These are all administrative: update ci infrastructure, fix tests, update libraries used and change the name of a provided library from nbd-lwt-unix to nbd-unix.

**Review by lindig — APPROVED** [review]

**Pull request merged**